src/kernels/api/api.ts

```diff
@@ -1,25 +1,52 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import { Uri } from 'vscode';
+import { Uri, extensions, window } from 'vscode';
 import { Kernel, Kernels } from '../../api';
 import { ServiceContainer } from '../../platform/ioc/container';
 import { IKernel, IKernelProvider, isRemoteConnection } from '../types';
 import { IVSCodeNotebook } from '../../platform/common/application/types';
 import { createKernelApiForExetnsion } from './kernel';
 
 // Each extension gets its own instance of the API.
-const apiCache = new Map<string, Kernels>();
+const apiCache = new Map<string, Promise<Kernels>>();
 const kernelCache = new WeakMap<IKernel, Kernel>();
 
-export function getKernelsApi(extensionId: string): Kernels {
-    // Each extension gets its own instance of the API.
-    let api = apiCache.get(extensionId);
-    if (api) {
-        return api;
+const allowedAccessToProposedApi = new Set(['ms-toolsai.datawrangler']);
+
+export async function requestKernelAccess(extensionId: string): Promise<Kernels> {
+    const promise = apiCache.get(extensionId) || requestKernelAccessImpl(extensionId);
+    apiCache.set(extensionId, promise);
+    return promise;
+}
+export async function requestKernelAccessImpl(extensionId: string): Promise<Kernels> {
+    if (!allowedAccessToProposedApi.has(extensionId)) {
+        throw new Error(`Extension ${extensionId} does not have access to proposed API`);
     }
+    const displayName = extensions.getExtension(extensionId)?.packageJSON?.displayName || extensionId;
+    // Not localized for now, as no one can use this except us (& DW).
+    // Will work on a formal API access separately.
+    const result = await window.showInformationMessage(
```

Comment:
Temporary, the proper solution will be done as part of https://github.com/microsoft/vscode-jupyter/issues/14685